### PR TITLE
enhance(toc): localize "In this article" header

### PR DESCRIPTION
**Pull request by caugner**
> [!IMPORTANT]  
> Currently, we do **not accept any community PRs for UI localization**.
> We will revisit UI localization later this year or early next year. Thank you for your understanding!

## Summary

(MP-1194)

### Problem

Our user interface isn't localized, but some content UI parts should be localized like macros.

### Solution

Localize the "In this article" header that is shown above the TOC.

---

### Before

image

### After

image

---

## How did you test this change?

Ran `yarn dev` and checked http://localhost:3000/fr/docs/Web/HTML locally.

### CODE REVIEW

File: client/src/document/organisms/toc/index.tsx [hunk 1]
```diff
@@ -5,8 +5,23 @@ import { Toc } from "../../../../../libs/types/document";
...
+const DEFAULT_TITLE = {
+  "en-US": "In this article",
+  es: "En este artículo",
+  fr: "Dans cet article",
+  ja: "この記事では",
+  ko: "이 문서에서는",
+  "pt-BR": "Neste artigo",
+  ru: "В этой статье",
+  "zh-CN": "在本文中",
+  "zh-TW": "在本文中",
```
Comment (caugner) [inline client/src/document/organisms/toc/index.tsx:20]: @mdn/yari-content-zh Would you mind taking a look if these translations are correct?

PS: Please note that this is a one-off UI localization, and currently we do not accept any community PRs for UI localization.

File: client/src/document/organisms/toc/index.tsx [hunk 2]
```diff
@@ -5,8 +5,23 @@ import { Toc } from "../../../../../libs/types/document";
...
+import { DEFAULT_LOCALE } from "../../../../../libs/constants";
+
+const DEFAULT_TITLE = {
+  "en-US": "In this article",
+  es: "En este artículo",
+  fr: "Dans cet article",
+  ja: "この記事では",
+  ko: "이 문서에서는",
+  "pt-BR": "Neste artigo",
+  ru: "В этой статье",
```
Comment (caugner) [inline client/src/document/organisms/toc/index.tsx:18]: @mdn/yari-content-ru Would you mind taking a look if this translation is correct?

PS: Please note that this is a one-off UI localization, and currently we do not accept any community PRs for UI localization.

File: client/src/document/organisms/toc/index.tsx [hunk 3]
```diff
@@ -5,8 +5,23 @@ import { Toc } from "../../../../../libs/types/document";
...
+import { useLocale } from "../../../hooks";
+import { DEFAULT_LOCALE } from "../../../../../libs/constants";
+
+const DEFAULT_TITLE = {
+  "en-US": "In this article",
+  es: "En este artículo",
+  fr: "Dans cet article",
+  ja: "この記事では",
+  ko: "이 문서에서는",
+  "pt-BR": "Neste artigo",
```
Comment (caugner) [inline client/src/document/organisms/toc/index.tsx:17]: @mdn/yari-content-pt-br Would you mind taking a look if this translation is correct?

PS: Please note that this is a one-off UI localization, and currently we do not accept any community PRs for UI localization.

File: client/src/document/organisms/toc/index.tsx [hunk 4]
```diff
@@ -5,8 +5,23 @@ import { Toc } from "../../../../../libs/types/document";
...
 import { TOC_CLICK } from "../../../telemetry/constants";
+import { useLocale } from "../../../hooks";
+import { DEFAULT_LOCALE } from "../../../../../libs/constants";
+
+const DEFAULT_TITLE = {
+  "en-US": "In this article",
+  es: "En este artículo",
+  fr: "Dans cet article",
+  ja: "この記事では",
+  ko: "이 문서에서는",
```
Comment (caugner) [inline client/src/document/organisms/toc/index.tsx:16]: @mdn/yari-content-ko Would you mind taking a look if this translation is correct?

PS: Please note that this is a one-off UI localization, and currently we do not accept any community PRs for UI localization.

File: client/src/document/organisms/toc/index.tsx [hunk 5]
```diff
@@ -5,8 +5,23 @@ import { Toc } from "../../../../../libs/types/document";
...
 import { useGleanClick } from "../../../telemetry/glean-context";
 import { TOC_CLICK } from "../../../telemetry/constants";
+import { useLocale } from "../../../hooks";
+import { DEFAULT_LOCALE } from "../../../../../libs/constants";
+
+const DEFAULT_TITLE = {
+  "en-US": "In this article",
+  es: "En este artículo",
+  fr: "Dans cet article",
+  ja: "この記事では",
```
Comment (caugner) [inline client/src/document/organisms/toc/index.tsx:15]: @mdn/yari-content-ja Would you mind taking a look if this translation is correct?

PS: Please note that this is a one-off UI localization, and currently we do not accept any community PRs for UI localization.

File: client/src/document/organisms/toc/index.tsx [hunk 6]
```diff
@@ -5,8 +5,23 @@ import { Toc } from "../../../../../libs/types/document";
 import { useFirstVisibleElement } from "../../hooks";
 import { useGleanClick } from "../../../telemetry/glean-context";
 import { TOC_CLICK } from "../../../telemetry/constants";
+import { useLocale } from "../../../hooks";
+import { DEFAULT_LOCALE } from "../../../../../libs/constants";
+
+const DEFAULT_TITLE = {
+  "en-US": "In this article",
+  es: "En este artículo",
+  fr: "Dans cet article",
```
Comment (caugner) [inline client/src/document/organisms/toc/index.tsx:14]: @mdn/yari-content-fr Would you mind taking a look if this translation is correct?

PS: Please note that this is a one-off UI localization, and currently we do not accept any community PRs for UI localization.

File: client/src/document/organisms/toc/index.tsx [hunk 7]
```diff
@@ -5,8 +5,23 @@ import { Toc } from "../../../../../libs/types/document";
 import { useFirstVisibleElement } from "../../hooks";
 import { useGleanClick } from "../../../telemetry/glean-context";
 import { TOC_CLICK } from "../../../telemetry/constants";
+import { useLocale } from "../../../hooks";
+import { DEFAULT_LOCALE } from "../../../../../libs/constants";
+
+const DEFAULT_TITLE = {
+  "en-US": "In this article",
+  es: "En este artículo",
```
Comment (caugner) [inline client/src/document/organisms/toc/index.tsx:13]: @mdn/yari-content-es Would you mind taking a look if this translation is correct?

PS: Please note that this is a one-off UI localization, and currently we do not accept any community PRs for UI localization.

File: client/src/document/organisms/toc/index.tsx (hunk 1 above)
Comment (yin1999) [reply client/src/document/organisms/toc/index.tsx:20]: LGTM. Thank you @caugner!

**Review by yin1999 — APPROVED** [review]

File: client/src/document/organisms/toc/index.tsx (hunk 7 above)
Comment (JuanVqz) [reply client/src/document/organisms/toc/index.tsx:13]: Was about to suggest adding the accent at `í` of Articulo, but it is already there, so, looks good! thanks! cc @Graywolf9

**Review by JuanVqz — APPROVED** [review]
From ES team, it looks good!

File: client/src/document/organisms/toc/index.tsx (hunk 4 above)
Comment (1ilsang) [reply client/src/document/organisms/toc/index.tsx:16]: image

@mdn/yari-content-ko  "목차"로 하는게 어떨까요?

File: client/src/document/organisms/toc/index.tsx (hunk 2 above)
Comment (leon-win) [reply client/src/document/organisms/toc/index.tsx:18]: Yes, it is good for `ru`, Thank you, @caugner!

**Review by leon-win — APPROVED** [review]

File: client/src/document/organisms/toc/index.tsx (hunk 1 above)
Comment (caugner) [reply client/src/document/organisms/toc/index.tsx:20]: Thank you @yin1999.

File: client/src/document/organisms/toc/index.tsx (hunk 4 above)
Comment (caugner) [reply client/src/document/organisms/toc/index.tsx:16]: @1ilsang @hochan222 Do you want to change it as follows:

```suggestion
  ko: "목차"로 하는게 어떨까요",
```

File: client/src/document/organisms/toc/index.tsx (hunk 6 above)
Comment (caugner) [reply client/src/document/organisms/toc/index.tsx:14]: From my point of view as a French speaker (not native), the translation looks good.

File: client/src/document/organisms/toc/index.tsx (hunk 5 above)
Comment (caugner) [reply client/src/document/organisms/toc/index.tsx:15]: @mdn/yari-content-ja Do you have any comments about this translation? 🙏

File: client/src/document/organisms/toc/index.tsx (hunk 3 above)
Comment (caugner) [reply client/src/document/organisms/toc/index.tsx:17]: @mdn/yari-content-pt-br Do you have any comments about this translation? 🙏

File: client/src/document/organisms/toc/index.tsx (hunk 2 above)
Comment (caugner) [reply client/src/document/organisms/toc/index.tsx:18]: Thank you @leon-win!

**Review by cw118 — APPROVED** [review]
Looks good for fr! Sorry for the delay

File: client/src/document/organisms/toc/index.tsx (hunk 4 above)
Comment (hochan222) [reply client/src/document/organisms/toc/index.tsx:16]: I agree :)

**Review by 1ilsang — APPROVED** [review]
Looks good for ko. Thanks!

**Review by fiji-flo — APPROVED** [review]

File: client/src/document/organisms/toc/index.tsx (hunk 5 above)
Comment (mfuji09) [reply client/src/document/organisms/toc/index.tsx:15]: @caugner
I am noticed this now. I'm sorry for too late reply.
"この記事では" looks odd, and "この記事内" looks better.

**Pull request merged**